File: 김윤재/Chapter6/2579.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n,input;
    vector<int> stairs;
    int score[301]={0};
    cin>>n; //n=계단수
    for (int i = 0; i<n; i++){cin>>input;stairs.push_back(input);}
    //stairs 인덱스가 하나씩 작음
    score[1]=stairs[0];
    score[2]=stairs[0]+stairs[1];
    for(int i = 3; i<=n; i++){
        score[i]=max(score[i-2]+stairs[i-1],score[i-3]+stairs[i-1]+stairs[i-2]);
    }
    
    cout<<score[n];
    return 0;
}

File: 최진영/Chapter7/2579.py
n = int(input())
point = [int(input()) for i in range(n)]
step = [point[0],point[0]+point[1],max(point[0],point[1])+point[2]]
for i in range(3,n): step.append(max(step[i-2],step[i-3]+point[i-1])+point[i])
print(step[-1])
